modules/core/src/main/java/org/apache/ignite/internal/cache/query/index/sorted/keys/BytesCompareUtils.java

```diff
@@ -19,41 +19,24 @@
 
 /** */
 public final class BytesCompareUtils {
-    /** */
-    public static int compareNotNullSigned(byte[] arr0, byte[] arr1) {
-        if (arr0 == arr1)
-            return 0;
-        else {
-            int commonLen = Math.min(arr0.length, arr1.length);
-
-            for (int i = 0; i < commonLen; ++i) {
-                byte b0 = arr0[i];
-                byte b1 = arr1[i];
-
-                if (b0 != b1)
-                    return b0 > b1 ? 1 : -1;
-            }
-
-            return Integer.signum(arr0.length - arr1.length);
-        }
-    }
-
     /** */
     public static int compareNotNullUnsigned(byte[] arr0, byte[] arr1) {
         if (arr0 == arr1)
             return 0;
         else {
             int commonLen = Math.min(arr0.length, arr1.length);
+            int unSignArr0;
+            int unSignArr1;
 
             for (int i = 0; i < commonLen; ++i) {
-                int unSignArr0 = arr0[i] & 255;
-                int unSignArr1 = arr1[i] & 255;
+                unSignArr0 = arr0[i] & 255;
+                unSignArr1 = arr1[i] & 255;
 
                 if (unSignArr0 != unSignArr1)
                     return unSignArr0 > unSignArr1 ? 1 : -1;
             }
 
-            return Integer.signum(arr0.length - arr1.length);
+            return Integer.signum(Integer.compare(arr0.length, arr1.length));
```

Comment:
One more function call. Why? A trivial operator might be slightly faster.

Reply:
`compare` contract doen't guarantee that only `-1, 0, 1` will be returned.
But logic inside index subsystem distinguish -1 and -2 results of comparison.
`Integer.signum` required to make sure that only expected values returned.